src/tribler/core/socks5/udp_connection.py

```diff
@@ -39,23 +39,32 @@ def get_listen_port(self) -> int:
         """
         Retrieve the listen port for this protocol.
         """
-        _, port = self.transport.get_extra_info("sockname")
+        if self.transport:
+            _, port = self.transport.get_extra_info("sockname")
+        else:
+            port = 0
         return port
 
     def send_datagram(self, data: bytes) -> bool:
         """
         Send a datagram to the known remote address. Returns False if there is no remote yet.
         """
         if self.remote_udp_address:
-            self.transport.sendto(data, self.remote_udp_address)
+            cast(DatagramTransport, self.transport).sendto(data, self.remote_udp_address)
             return True
         self._logger.error("cannot send data, no clue where to send it to")
         return False
 
-    def datagram_received(self, data: bytes, source: tuple) -> bool:
+    def datagram_received(self, data: bytes, source: tuple) -> None:
         """
         The callback for when data is handed to our protocol.
         """
+        self.datagram_received(data, source)
```

Comment:
This doesn't look right. I'm guessing this should be calling `datagram_received`.

Reply:
Thanks! Fixed. 👌